server/src/endpoints/forward.rs

```diff
@@ -41,65 +41,68 @@ fn get_forwarded_for<S>(request: &HttpRequest<S>) -> String {
-        .uri()
-        .path_and_query()
-        .map(|pq| pq.as_str())
-        .unwrap_or("");
+        let path_and_query = request
+            .uri()
+            .path_and_query()
+            .map(|pq| pq.as_str())
+            .unwrap_or("");
 
-    let mut forwarded_request_builder = ClientRequest::build();
-    for (key, value) in request.headers() {
-        // Since there is no API in actix-web to access the raw, not-yet-decompressed stream, we
-        // must not forward the content-encoding header, as the actix http client will do its own
-        // content encoding. Also remove content-length because it's likely wrong.
-        if HOP_BY_HOP_HEADERS.contains(key) || IGNORED_REQUEST_HEADERS.contains(key) {
-            continue;
-        }
-        forwarded_request_builder.header(key.clone(), value.clone());
-    }
-    forwarded_request_builder
-        .no_default_headers()
-        .disable_decompress()
-        .method(request.method().clone())
-        .uri(upstream.get_url(path_and_query))
-        .set_header("Host", upstream.host())
-        .set_header("X-Forwarded-For", get_forwarded_for(request))
-        .set_header("Connection", "close");
-
-    ForwardBody::new(request)
-        .limit(config.max_api_payload_size())
-        .map_err(Error::from)
-        .and_then(move |data| forwarded_request_builder.body(data).map_err(Error::from))
-        .and_then(move |request| request.send().map_err(Error::from))
-        .and_then(move |response| {
-            let mut forwarded_response = HttpResponse::build(response.status());
-            // For the response body we're able to disable all automatic decompression and
-            // compression done by actix-web or actix' http client.
-            //
-            // 0. use ClientRequestBuilder::disable_decompress() (see above)
-            //
-            // 1. Set content-encoding to identity such that actix-web will not attempt to compress
-            //    again
-            forwarded_response.content_encoding(ContentEncoding::Identity);
-
-            for (key, value) in response.headers() {
-                // 2. Just pass content-length, content-encoding etc through
-                if HOP_BY_HOP_HEADERS.contains(key) {
-                    continue;
-                }
-                forwarded_response.header(key.clone(), value.clone());
+        let mut forwarded_request_builder = ClientRequest::build();
+        (for (key, value) in request.headers() {
+            // Since there is no API in actix-web to access the raw, not-yet-decompressed stream, we
+            // must not forward the content-encoding header, as the actix http client will do its own
+            // content encoding. Also remove content-length because it's likely wrong.
+            if HOP_BY_HOP_HEADERS.contains(key) || IGNORED_REQUEST_HEADERS.contains(key) {
+                continue;
+            }
+            forwarded_request_builder.header(key.clone(), value.clone());
+        });
```

Comment:
@mitsuhiko rustfmt now works inside of that macro, except that I am apparently not able to match against this:

    for _ in .. { .. }

It has to have a trailing semicolon:

    for _ in .. { .. };

However, rustfmt will remove that semicolon, so I just added another pair of parens around the for-loop to prevent that:

    (for _ in .. { });